File: VWWPermissionKitExample/ViewController.m
//
//  ViewController.m
//  VWWPermissionKitExample
//
//  Created by Zakk Hoyt on 6/13/15.
//  Copyright (c) 2015 Zakk Hoyt. All rights reserved.
//

#import "ViewController.h"
#import <VWWPermissionKit/VWWPermissionKit.h>

@implementation ViewController

-(void)viewDidAppear:(BOOL)animated{
    [super viewDidAppear:animated];
    //    [self setupAppearance];
    [self showPermissions];
}

-(void)setupAppearance{
    [[UILabel appearance] setTextColor:[UIColor cyanColor]];
    [[UILabel appearance] setFont:[UIFont fontWithName:@"Marker Felt" size:14]];
    [[UIButton appearance] setTintColor:[UIColor magentaColor]];
    [[UIButton appearance] setTitleColor:[UIColor orangeColor] forState:UIControlStateNormal];
    [[UIButton appearance].titleLabel setFont:[UIFont fontWithName:@"Marker Felt" size:20]];
}

-(void)showPermissions{
    VWWContactsPermission *contacts = [VWWContactsPermission permissionWithLabelText:@"The app will merge duplicate contacts."];
    VWWPhotosPermission *photos = [VWWPhotosPermission permissionWithLabelText:@"We can save recorded videos to your Photos library."];
    VWWCoreLocationAlwaysPermission *locationAlways = [VWWCoreLocationAlwaysPermission permissionWithLabelText:@"For calculating your heading, altitude, speed, distance home, and more."];
    
    NSArray *permissions = @[contacts, locationAlways, photos];

    
    [VWWPermissionsManager optionPermissions:permissions
                                       title:@"Welcome! To get the most out of this Fancy App, we will need your permission to use some things. Let's get started."
                          fromViewController:self
                                resultsBlock:^(NSArray *permissions) {
                                    [permissions enumerateObjectsUsingBlock:^(VWWPermission *permission, NSUInteger idx, BOOL *stop) {
                                        NSLog(@"%@ - %@", permission.type, [permission stringForStatus]);
                                    }];
                                }];
}

@end

File: VWWPermissionKitExampleSwift/ViewController.swift
//
//  ViewController.swift
//  VWWPermissionKitExampleSwift
//
//  Created by Zakk Hoyt on 6/15/15.
//  Copyright (c) 2015 Zakk Hoyt. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)
        setupAppearance()
        showPermissions()
    }
    
    func setupAppearance() {
        UILabel.appearance().textColor = UIColor.cyanColor()
        UILabel.appearance().font = UIFont(name: "Marker Felt", size: 14)
        UIButton.appearance().tintColor = UIColor.magentaColor()
        UIButton.appearance().setTitleColor(UIColor.orangeColor(), forState: .Normal)
        UIButton.appearance().titleLabel!.font = UIFont(name: "Marker Felt", size: 20)
    }
    
    func showPermissions() {
        let photos = VWWPhotosPermission.permissionWithLabelText("In order to write to your Camera Roll")
        let camera = VWWCameraPermission.permissionWithLabelText("In order to access your camera to record video.")
        let coreLocationAlways = VWWCoreLocationAlwaysPermission.permissionWithLabelText("To calculate your heading, altitude, speed, distance home, etc...")
        
        let shareTypes = [HKObjectType.quantityTypeForIdentifier(HKQuantityTypeIdentifierBloodAlcoholContent),
            HKObjectType.quantityTypeForIdentifier(HKQuantityTypeIdentifierRespiratoryRate),
            HKObjectType.quantityTypeForIdentifier(HKQuantityTypeIdentifierHeartRate)] as [AnyObject]
        let readTypes = shareTypes
        let health: VWWHealthPermission = VWWHealthPermission.permissionWithLabelText("")
//        let health: VWWHealthPermission = VWWHealthPermission.permissionWithLabelText("test", shareTypes: shareTypes, readTypes: readTypes)
        
        let permissions = [photos, camera, coreLocationAlways, health]
        
        // Using requirePermissions:permissions, the user cannot proceed until all permissions are authorized
        VWWPermissionsManager.requirePermissions(permissions, title: "We need your approvoal before we get running", fromViewController: self) { (permissions: [AnyObject]!) -> Void in
            print("permission")
        }
        
    }
}

